pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -586,6 +607,16 @@ func (r *ReconcileBareMetalHost) actionProvisioning(prov provisioner.Provisioner
 		return actionContinueNoWrite{}
 	}
 
+	// Ensure the root device hints we're going to use are stored.
+	dirty, err := updateRootDeviceHintsForHost(info.host)
```

Comment:
If we move this call to https://github.com/metal3-io/baremetal-operator/blob/df37555045c5d10bad0b2b83b471a9c4951bbe4d/pkg/controller/baremetalhost/baremetalhost_controller.go#L813 then the change will be atomic with the state transition from Ready->Provisioning, thus guaranteeing that it can't change under us. I'm happy to submit a separate PR for that, but I can't think of any reason not to go ahead and do it here.

Reply:
good point, done